src/components/chip/selectable-chip/index.tsx

```diff
@@ -47,29 +48,41 @@ export const SelectableChip = forwardRef<
     ) => {
       onDismiss?.(event)
     }
-
     return (
       <button
         className={chipBaseVariant({
           dismissible,
           size,
           className: qtMerge(
             chipBaseClassnames,
-            chipSelectionClassnames,
+            chipBaseTextColorClassnames,
             className,
           ),
         })}
-        data-state=""
+        data-state={dropdownItemSelected ? 'selected' : ''}
         ref={ref}
         onClick={handleClick}
         {...rest}
       >
-        {Icon && <Icon {...ChipIconSizes[size]} />}
-        {children}
-        {labelTag && <span className="font-bold">{labelTag}</span>}
+        {Icon && <Icon id="selectable-chip-icon" {...ChipIconSizes[size]} />}
+        <div
+          id="selectable-chip-label"
+          className="transition-transform duration-1000 ease-in-out"
+        >
+          {children}
+        </div>
+        {labelTag && (
+          <span
+            id="selectable-chip-label-tag"
+            className={`font-bold ${chipBaseTextColorClassnames}`}
+          >
+            {labelTag}
+          </span>
+        )}
         {dismissible && (
           <StandaloneCircleXmarkRegularIcon
             {...ChipStandaloneIconSizes[size]}
+            id="selectable-chip-dismiss-icon"
```

Comment:
@shafin-deriv, can this `id="selectable-chip-dismiss-icon"` value be dynamically passed in through props?

I think the `id` might conflict if you have more than one selectable chip on a page. You could try to use css combinators to target the icon in case the `ids` conflict if multiple chips are used on a page

Reply:
@jim-deriv I didn't think about this way, but definitely a valid point. I just wanted to put the id just in case if someone need it for any test purpose or any other thing. I think it's better if I just remove the id. At the parent we are already allowing to pass id (I mean ...rest) and if anyone needs to access this DOM node I think they can do so by using that only. Thanks for raising this!